File: Language specifics/Language-specifics-Java Script/CompareCharArrays.js
function compare(input) {
    let chars1 = input[0].split(" ");
    let chars2 = input[1].split(" ");

    if(chars1.length > chars2.length){
        console.log(chars2.join(""));
        console.log(chars1.join(""));
    }
    else if(chars1.length<chars2.length){
        console.log(chars1.join(""));
        console.log(chars2.join(""));
    }
    else{
        for (let i=0; i<chars1.length; i++){
            if(chars1[i] > chars2[i]){
                console.log(chars2.join(""));
                console.log(chars1.join(""));
                break;
            }
            else if(chars1[i] < chars2[i]){
                console.log(chars1.join(""));
                console.log(chars2.join(""));
                break;
            }
            else{
                if(i === chars1.length-1) {
                    console.log(chars1.join(""));
                    console.log(chars2.join(""));
                }
            }
        }
    }
}
//compare(["a n n","a n n a"]);
//compare(["a n n a","a n"]);
//compare(["a n n","b a b"]);
compare(["a a a a a a a a a a a a a a","a a a a a a a a a a a a a b"]);

File: Language specifics/Language-specifics-Java/src/CompareCharArrays.java
import java.util.Arrays;
import java.util.Scanner;

public class CompareCharArrays {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        String[] chars1 = scan.nextLine().split(" ");
        String[] chars2 = scan.nextLine().split(" ");

        if(chars1.length > chars2.length){
            System.out.println(String.join("",  chars2));
            System.out.println(String.join("", chars1));
        }
        else if(chars1.length < chars2.length){
            System.out.println(String.join("", chars1));
            System.out.println(String.join("",  chars2));
        }
        else{
            for(int i = 0; i < chars1.length; i++){
                if(chars1[i].equals(chars2[i])){
                    if(i == chars1.length - 1){
                        System.out.println(String.join("",  chars1));
                        System.out.println(String.join("", chars2));
                    }
                }
                else if(chars1[i].compareTo(chars2[i]) < 0){
                    System.out.println(String.join("",  chars1));
                    System.out.println(String.join("", chars2));
                    break;
                }
                else if(chars1[i].compareTo(chars2[i]) > 0){
                    System.out.println(String.join("",  chars2));
                    System.out.println(String.join("", chars1));
                    break;
                }
            }
        }
    }
}

